# Aluno: Diogo Luiz de Oliveira Coelho

3) (3,0	PONTOS)	Considere um meio constituído por duas camada de espessura $h_{1}$ = 12 km e velocidade de onda P $\alpha _{1}$ = 7 km/s e de espessura $h_{2}$ = 18 km e velocidade de onda P $\alpha _{2}$= 6 km/s, respectivamente, acima de um semi-espaço de velocidade de onda P $\alpha _{3}$ = 8 km/s. Há uma fonte sísmica a uma profundidade de $h$ = 6 km abaixo da superfície. Para uma estação localizada a 100 km de distância epicentral, calcular:

![caption](images/esquema_3)


(i) O tempo de percurso da onda P refletida na base da 1ª camada.

(ii) O tempo de percurso da onda P refletida na base da 2ª camada.

(iii) O tempo de percurso da onda refratada criticamente.

In [1]:
import numpy as np

R:
---

## (i) O tempo de percurso da onda P refletida na base da 1ª camada.

Utilizando a Figura original traçou-se o raio de uma onda P incidindo sobre uma camada horizontal e sua posterior reflexão.

![caption](images/reflect_wave)


Segundo o esquema acima pode-se calcular o percurso da onda P($P_{r1}$) refletida na base da 1ª camada através do teorema de pitágoras:


(cateto oposto) ² + (cateto adjacente) ² = (Hipotenusa) ² 

$(h_{1}-h+h_{1}-h+h)^2 + (x)^2 = (P_{r1})²$

$(2h_{1}-h)^2 + (x)^2 = (P_{r1})²$

$P_{r1} = \sqrt{(2h_{1}-h)^2 + (x)^2}$


Sendo $h_{1} = 12 km $, $h = 6 km$ e $X = 100 km $, temos:  

$P_{r1} = \sqrt{(2.12-6)^2 + (100)^2}$

In [2]:
np.sqrt((2*12-6)**2 + (100)**2)

101.6070863670443

Para o cálculo do tempo de percurso da onda P refletida ($T_{r1}$) na base da 1ª camada é feita a divisão do percurso pela velocidade da onda P na 1ª camada:

$T_{r1} = \frac{P_{r1}}{\alpha _{1}} $

sendo $\alpha _{1} = 7 km/s$

In [3]:
101.6070863670443/7


14.5152980524349

### Portanto tempo de percurso da onda P refletida ($T_{r1}$) é 14.51 segundos.

____________________________________________

## (ii) O tempo de percurso da onda P refletida na base da 2ª camada.

Utilizando a Figura original traçou-se o raio de uma onda P incidindo sobre a 1ª camada horizontal e sendo refratada, logo após incide sobre a 2ª camada e é refletida.

![caption](images/reflect_wave_2)

Devido ao grande número de incógnitas, como $i_{1}$,$i_{1}$, $A$,$B$, $C$ e $D$, utilizou-se relações trigonométricas para montar um sistema capaz de resolver estas incógnitas. Com isso temos que :

$cos i_{1} = \frac{(h_{1}-h)}{A}$ (1)

$sin i_{2} = \frac{X_{1}/2}{B}$ (2)

$sin i_{2} = \frac{X_{1}/2}{C}$ (3)

$cos i_{1} = \frac{h_{1}}{D}$ (4)

Pela Lei de Snell temos:

$\frac{sin i_{1}}{\alpha _{1}} = \frac{sin i_{2}}{\alpha _{2}}$ (5)

Pela Lei dos cossenos temos:

$X_{1}^2 = B^2 + C^2 - 2 B . C. cos (2i_{2})$ (6)

Pela semelhança de triângulos podemos ver que $B$ = $C$ (7).

Pelo teorema de pitágoras teremos:

$(X_{1}/2)^2 = B^2 + h_{2}^2$ (8)

Substituindo (7) em (6) temos:

$X_{1}^2 = B^2 + B^2 - 2 B . B. cos (2i_{2})$ 

$X_{1}^2 = 2B^2 - 2 B^2. cos (2i_{2})$

$X_{1}^2 = 2B^2(1 - cos (2i_{2}))$

Utilizando a identidade trigonométrica que $cos (2i_{2}) = 2cos^2 (i_{2}) - 1)$ temos:

$X_{1}^2 = 2B^2(1 - 2cos^2 (i_{2}) - 1))$

$X_{1}^2 = 2B^2(2cos^2 (i_{2}))$

$X_{1}^2 = 2B^2(2cos^2 (i_{2}))$

$X_{1}^2 = 4B^2cos^2 (i_{2}))$, simplificando temos:

### $X_{1} = 2Bcos (i_{2})$ 

Pela figura podemos encontrar que $cos i_{2} = \frac{h_{2}}{B}$, substituindo na equação acima teremos:

$X_{1} = 2B\frac{h_{2}}{B}$ 

### $X_{1} = 2h_{2}$

Pela equação (8) temos:

$B^2 = (X_{1}/2)^2 + h_{2}^2$, substituindo o valor de $X_{1}$:

$B^2 = (2h_{2}/2)^2 + h_{2}^2$:

$B^2 = h_{2}^2 + h_{2}^2$, então:

$B^2 = 2h_{2}^2$, simplificando:

$B = \sqrt{2}h_{2}$, portanto $C=\sqrt{2}h_{2}$.  

$B = \sqrt{2}*18$ e $C=\sqrt{2}*18$.  

In [4]:
np.sqrt(2)*18

25.455844122715714

### $B = 25.45 km$ e $C=25.45 km$.  

Calculando a equação (2):

$sin i_{2} = \frac{X_{1}/2}{B}$, substituindo os valores temos:

$sin i_{2} = \frac{2h_{2}}{2\sqrt{2}h_{2}}$

$sin i_{2} = \frac{1}{\sqrt{2}}$

$i_{2} = arcsin (\frac{1}{\sqrt{2}})$

In [5]:
1/np.sqrt(2)

0.70710678118654746

$i_{2} = arcsin (0.70710678118654746)$

$i_{2} = 45 ^{\circ}$

Pela lei de Snell (5) temos:

$\frac{sin i_{1}}{\alpha _{1}} = \frac{sin i_{2}}{\alpha _{2}}$ , substituindo os valores:

$\frac{sin i_{1}}{7} = \frac{sin 45 ^{\circ}}{6}$ 

$sin i_{1} = \frac{7sin 45 ^{\circ}}{6}$, calculando:

$sin i_{1} = \frac{7*0.70710678}{6}$, calculando:

In [6]:
7*0.70710678/6

0.8249579100000001

$sin i_{1} = 0.8249579100000001$

$i_{1} = arcsin(0.8249579100000001)$

$i_{1} = 55.58 ^{\circ}$

Calculando A através de (1):

$A = \frac{(h_{1}-h)}{cos i_{1}}$, substituindo:

$A = \frac{(12-6)}{cos (55.58 ^{\circ})}$

$A = \frac{6}{0.56525499}$

In [7]:
6/0.56525499

10.614678518804407

### $A = 10.61 km$

Calculando D através de (4):

$cos i = \frac{h_{1}}{D}$,substituindo:
  
$cos (55.58 ^{\circ}) = \frac{12}{D}$

$D = \frac{12}{0.56525499}$

In [8]:
12/0.56525499

21.229357037608814

### $D = 21.22 km$

Para calcular o tempo de percurso é necessário dividir cada trajeto pela velocidade de cada camada, e no final fazer o somatório dos tempos para encontrar o tempo total. 

$T_{r2} = \frac{A}{\alpha _{1}} + \frac{B}{\alpha _{2}} + \frac{C}{\alpha _{2}} + \frac{D}{\alpha _{1}} $, substituindo os valores de cada intervalo teremos:

$T_{r2} = \frac{10.61}{7} + \frac{25.45}{6} + \frac{25.45}{6} + \frac{21.22}{7} $, calculando:

In [16]:
10.61/7 + 25.45/6 + 25.45/6 + 21.22/7

13.03047619047619

### Portanto tempo de percurso da onda P refletida ($T_{r2}$) é 13.03 segundos.

## (iii) O tempo de percurso da onda refratada criticamente.

Através da Lei de Snell podemos calcular os ângulos da onda incidente na 1ª camada ($i$) e o ângulo crítico ($i_{c}$) da 2ª camada segundo o esquema abaixo:


![caption](images/critical_wave)


$\frac{sin i}{\alpha _{1}} = \frac{sin i_{c}}{\alpha _{2}} = \frac{1}{\alpha _{3}}$, 

sendo $\alpha _{1} = 7 km/s$, $\alpha _{2} = 6 km/s$ e $\alpha _{3} = 8km/s $.

Calculando o ângulo de incidência:

$\frac{sin i}{\alpha _{1}} = \frac{1}{\alpha _{3}}$

$\frac{sin i}{7} = \frac{1}{8}$

$\sin i = \frac{7}{8}$

$i = arcsin(\frac{7}{8})$

$i = arcsin(0.875)$

$i = 61.04^{\circ}$
__________

$\frac{sin i_{c}}{\alpha _{2}} = \frac{1}{\alpha _{3}}$

$\frac{sin i_{c}}{6} = \frac{1}{8}$

$\sin i_{c} = \frac{6}{8}$

$i_{c} = arcsin(\frac{6}{8})$

$i_{c} = arcsin(0.75)$

$i_{c} = 48.59 ^{\circ}$

_____________________________

Para se calcular o tempo de percurso para a onda P refratada criticamente ($T_{c}$) deve-se somar todos os trajetos informados na figura acima (a até e) e dividí-los pela velocidade de cada camada.

$T_{c} = \frac{A}{\alpha _{1}} + \frac{B}{\alpha _{2}} + \frac{C}{\alpha _{3}} + \frac{D}{\alpha _{2}} + \frac{E}{\alpha _{1}}$

Calculando cada segmento por trigonometria:

$cos i = \frac{h_{1}-h}{A}$

$A = \frac{h_{1}-h}{cos i}$

$A = \frac{12-6}{cos (61.04 ^{\circ})}$

$A = \frac{6}{0.4841989}$

In [10]:
6/0.4841989

12.391601880962556

### $A = 12.39 km$

$cos i_{c} = \frac{h_{2}}{B}$

$B = \frac{h_{2}}{cos i_{c}}$

$B = \frac{18}{cos (48.59 ^{\circ})}$

$B = \frac{18}{0.66144277}$

In [11]:
18/0.66144277

27.213238720562327

### $B = 27.21 km$
________________

Para o cálculo de $C$ fez-se a diferença entre a distância epicentral ($X$)
 e o segmento pontilhado vermelhor na figura, então:
 
$C = X - ([sin i.A] + [sin i_{c}.B] + [sin i_{c}.D] + [sin i.E])$

$C = 100 - ([sin(61.04^{\circ}) *12.39] + [sin (48.59^{\circ})*27.21] + [sin (48.59^{\circ})*27.21] + [sin (61.04^{\circ})*24.78])$

$C = 100 - ([0.87495795 *12.39] + [0.74999564*27.21] + [0.74999564*27.21] + [0.87495795*24.78])$

In [12]:
100-(0.87495795 *12.39 + 0.74999564*27.21 + 0.74999564*27.21 + 0.87495795*24.78)

26.66305026970001

### C = 26.66 km
__________________

Por semelhança de triângulos vê-se que $B$ é igual a $ D$, logo:
### $D = 27.21 km$.
____________

$cos i = \frac{h_{1}}{E}$

$E = \frac{h_{1}}{cos i}$

$E = \frac{12}{cos (61.04 ^{\circ})}$

$E = \frac{12}{0.4841989}$

In [13]:
12/0.4841989

24.78320376192511

### $E = 24.78 km$.
________


Substituindo os valores dos percursos na equação para calcular o tempo de percurso temos:

$T_{c} = \frac{A}{\alpha _{1}} + \frac{B}{\alpha _{2}} + \frac{C}{\alpha _{3}} + \frac{D}{\alpha _{2}} + \frac{E}{\alpha _{1}}$

$T_{c} = \frac{12.39}{7} + \frac{27.21}{6} + \frac{26.66 }{8} + \frac{27.21}{6} + \frac{24.78}{7}$

In [14]:
12.39/7 + 27.21/6 + 26.66/8 + 27.21/6 + 24.78/7

17.7125

O tempo de percurso da onda refratada criticamente é 
## $T_{c} = 17.71 s$